core/src/test/java/org/apache/iceberg/TestFastAppend.java

```diff
@@ -42,6 +42,25 @@ protected static List<Object> parameters() {
     return Arrays.asList(1, 2, 3);
   }
 
+  @TestTemplate
+  public void testAddManyFiles() {
+    assertThat(listManifestFiles()).as("Table should start empty").isEmpty();
+
+    List<DataFile> dataFiles = Lists.newArrayList();
+
+    for (int ordinal = 0; ordinal < 20_000; ordinal++) {
```

Comment:
nit: change `20_000` to `MIN_FILE_GROUP_SIZE * 2`. 

applicable to other tests.

Reply:
This would require making `MIN_FILE_GROUP_SIZE` package-private. I was debating that.
What's your opinion, @stevenzwu?

Reply:
I am usually good with package private. we have quite a bit `@VisibleForTesting` for case like that.

It can be tricky to update the tests if we ever change the `MIN_FILE_GROUP_SIZE` default value to some larger value like 20_000. then this test is not covering the intended case.

Reply:
Done.